src/internal/join-host-port.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * joinHostPort combines host and port into a network address of the
+ * form "host:port". If host contains a colon, as found in literal
+ * IPv6 addresses, then JoinHostPort returns "[host]:port".
+ *
+ * @param host
+ * @param port
+ * @returns Cleaned up host
+ * @internal
+ */
+export function joinHostPort(host: string, port?: number): string {
+  if (port === undefined) {
+    return host
+  }
+
+  // We assume that host is a literal IPv6 address if host has
+  // colons.
+  if (host.includes(':')) {
```

Comment:
While this may be sufficient but it is kludgy and can be faked easily. My suggestion is look for the total octets and parse it properly.

https://github.com/TritonDataCenter/node-ip6addr

Reply:
I don't think that parsing real IPv6 address is needed. On my implementation, I just ported Go's `net.JoinHostPort` function (code here https://cs.opensource.google/go/go/+/go1.21.4:src/net/ipsock.go;l=235). I think that is sufficient enough.

Reply:
Okay lets live with it for now.